lib/Parse/ParseType.cpp

```diff
@@ -901,6 +975,7 @@ bool Parser::canParseTypeIdentifier() {
   }
 }
 
+/// Can parse the deprecated 'protocol<...>'
 bool Parser::canParseTypeComposition() {
```

Comment:
Two notes:
- Doc comments go on the first declaration, not on the implementation.
- I think it would be better to rename this, something like `canParseOldStyleProtocolComposition`.